src/components/dialog/dialog.pw.tsx

```diff
@@ -0,0 +1,596 @@
+      mount,
+      page,
+    }) => {
+      await mount(<DialogComponent title={title} />);
+      await expect(page.getByText(title)).toBeAttached();
+    });
+  });
+
+  specialCharacters.forEach((subtitle) => {
+    test(`when subtitle prop is ${subtitle}, should use it as Dialog subtitle`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <DialogComponent title="Sample dialog" subtitle={subtitle} />
+      );
+      await expect(page.getByText(subtitle)).toBeAttached();
+    });
+  });
+
+  [
+    { size: SIZE.EXTRASMALL, width: "300px" },
+    { size: SIZE.SMALL, width: "380px" },
+    { size: SIZE.MEDIUMSMALL, width: "540px" },
+    { size: SIZE.MEDIUM, width: "750px" },
+    { size: SIZE.MEDIUMLARGE, width: "850px" },
+    { size: SIZE.LARGE, width: "960px" },
+    { size: SIZE.EXTRALARGE, width: "1080px" },
+  ].forEach(({ size, width }) => {
+    test(`when size prop is ${size}, Dialog width should be ${width}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<DialogComponent size={size} />);
+      await expect(page.getByRole("dialog")).toHaveCSS("width", width);
+    });
+  });
+
+  test("when showCloseIcon prop is false, should not render close icon", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent showCloseIcon={false} />);
+    await expect(page.getByLabel("Close")).not.toBeVisible();
+  });
+
+  test("when showCloseIcon prop is passed, clicking close icon closes Dialog", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent />);
+    const dialog = page.getByRole("dialog");
+    await expect(dialog).toBeVisible();
+
+    await page.getByLabel("Close").click();
+
+    await expect(dialog).not.toBeVisible();
+  });
+
+  test("pressing the Escape key closes Dialog", async ({ mount, page }) => {
+    await mount(<DialogComponent disableEscKey={false} />);
+    const dialog = page.getByRole("dialog");
+    await expect(dialog).toBeVisible();
+
+    await dialog.press("Escape");
+
+    await expect(dialog).not.toBeVisible();
+  });
+
+  test("when disableEscKey prop is passed, pressing the Escape key does not close Dialog", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent disableEscKey />);
+    const dialog = page.getByRole("dialog");
+    await expect(dialog).toBeVisible();
+
+    await dialog.press("Escape");
+
+    await expect(dialog).toBeVisible();
+  });
+
+  test("when a function is passed to the onCancel prop, clicking close icon should call that function", async ({
+    mount,
+    page,
+  }) => {
+    let called = false;
+    await mount(
+      <DialogComponent
+        onCancel={() => {
+          called = true;
+        }}
+      />
+    );
+
+    await page.getByLabel("Close").click();
+    expect(called).toBeTruthy();
+  });
+
+  test("when clicking outside the Dialog, the Dialog should not close", async ({
+    mount,
+    page,
+  }) => {
```

Comment:
suggestion: (non-blocking) it would be better to add a space to separate the mount from the assertion to follow the same pattern as with the other test cases.